api/client/import.go

```diff
@@ -78,3 +121,43 @@ func (cli *DockerCli) CmdImport(args ...string) error {
 
 	return jsonmessage.DisplayJSONMessagesStream(responseBody, cli.out, cli.outFd, cli.isTerminalOut)
 }
+
+func determineSourceAction(source string) int {
+	if source == "-" {
+		return srcStdin
+	}
+	if !urlutil.IsURL(source) {
+		return srcFile
+	}
+	return srcRemote
+}
+
+func verifyTarball(r io.Reader) error {
+	buff := make([]byte, 512)
+	if _, err := r.Read(buff); err != nil {
+		return err
+	}
+	contType := http.DetectContentType(buff)
+	r = io.MultiReader(bytes.NewReader(buff), r)
+	if contType == "application/x-gzip" {
+		var err error
+		r, err = gzip.NewReader(r)
+		if err != nil {
+			return err
+		}
+	}
+	tr := tar.NewReader(r)
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		if hdr.Name == "repositories" {
+			return fmt.Errorf("Error: the tarball is suspected in being an image due to the \"repositories\" file.")
```

Comment:
Wondering if we should make this a utility to determine the *type* of tarball. Then both *import* and *load* can check if it's the right type, and suggest to use the other command?

@aaronlehmann @unclejack @runcom wdyt?